exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/metrics/MetricStatelessMarshaler.java

```diff
@@ -0,0 +1,224 @@
+            return StatelessMarshalerUtil.sizeMessageWithContext(
+                dataField,
+                metricData.getHistogramData(),
+                HistogramStatelessMarshaler.INSTANCE,
+                context);
+          }
+
+          @Override
+          public void writeTo(Serializer output, MetricData metric, MarshalerContext context)
+              throws IOException {
+            output.serializeMessageWithContext(
+                Metric.HISTOGRAM,
+                metric.getHistogramData(),
+                HistogramStatelessMarshaler.INSTANCE,
+                context);
+          }
+        });
+    DATA_HANDLERS.put(
+        EXPONENTIAL_HISTOGRAM,
+        new DataHandler(Metric.EXPONENTIAL_HISTOGRAM) {
+          @Override
+          public int calculateSize(MetricData metricData, MarshalerContext context) {
+            return StatelessMarshalerUtil.sizeMessageWithContext(
+                dataField,
+                metricData.getExponentialHistogramData(),
+                ExponentialHistogramStatelessMarshaler.INSTANCE,
+                context);
+          }
+
+          @Override
+          public void writeTo(Serializer output, MetricData metric, MarshalerContext context)
+              throws IOException {
+            output.serializeMessageWithContext(
+                Metric.EXPONENTIAL_HISTOGRAM,
+                metric.getExponentialHistogramData(),
+                ExponentialHistogramStatelessMarshaler.INSTANCE,
+                context);
+          }
+        });
+  }
+
+  @Override
+  public void writeTo(Serializer output, MetricData metric, MarshalerContext context)
+      throws IOException {
+    DataHandler dataHandler = DATA_HANDLERS.get(metric.getType());
+    if (dataHandler == null) {
+      // Someone not using BOM to align versions as we require. Just skip the metric.
+      return;
+    }
+
+    output.serializeStringWithContext(Metric.NAME, metric.getName(), context);
+    output.serializeStringWithContext(Metric.DESCRIPTION, metric.getDescription(), context);
+    output.serializeStringWithContext(Metric.UNIT, metric.getUnit(), context);
+
+    dataHandler.writeTo(output, metric, context);
+  }
+
+  @Override
+  public int getBinarySerializedSize(MetricData metric, MarshalerContext context) {
+    DataHandler dataHandler = DATA_HANDLERS.get(metric.getType());
+    if (dataHandler == null) {
+      // Someone not using BOM to align versions as we require. Just skip the metric.
+      return 0;
+    }
+
+    int size = 0;
+    size += StatelessMarshalerUtil.sizeStringWithContext(Metric.NAME, metric.getName(), context);
+    size +=
+        StatelessMarshalerUtil.sizeStringWithContext(
+            Metric.DESCRIPTION, metric.getDescription(), context);
+    size += StatelessMarshalerUtil.sizeStringWithContext(Metric.UNIT, metric.getUnit(), context);
+
+    size += dataHandler.calculateSize(metric, context);
+
+    return size;
+  }
+
+  private abstract static class DataHandler {
```

Comment:
This is the same interface as `StatelessMarshaler`. Might as well reuse that. Should be reasonable to drop this abstract class and refactor to use `StatelessMarshaler<MetricData>` everywhere DataHandler is used.